### PR TITLE
Technical debt issues (see #13)

### PR DESCRIPTION
**Pull request by ggiraldez**
Ongoing effort to reduce technical debt

- console.log in debounced actions
- reduce boilerplate in redux actions (splitting constants, using objects instead of switch): NOT DONE due to losing Flow type checking for reducers:
  - it loses type specificity for imported constants
  - cannot do type refinement when choosing the reduce handler from a map (as opposed to using if/switch)
- explicit action creator functions
- more XLS form validations
- moved type definitions for translations and stats into schema

### CODE REVIEW

File: app/javascript/actions/bots.js [hunk 1]
```diff
@@ -8,40 +8,45 @@ export const RECEIVE = 'BOTS_RECEIVE'
...
+  type: CREATE_SUCCESS,
+  bot
+})
+
 export const fetchBots = () => (dispatch : T.Dispatch, getState : T.GetState) => {
   const state = getState()
 
   // Don't fetch Bots if they are already being fetched
-  if ((state.bots.fetching : boolean)) {
+  if (state.bots.fetching) {
```
Comment (bcardiff) [inline app/javascript/actions/bots.js]: 👎  unless `( ... : boolean)` is used it wont detect typos/errors in outer property in guards due to undefined and falsey semantics of javascript.

Alternative `( ... : *)` could be used to force some type.

Ref: https://github.com/facebook/flow/issues/3887

File: app/javascript/actions/bot.js [hunk 2]
```diff
@@ -6,40 +6,71 @@ import { pushNotification } from './notifications'
 
 export const UPDATE = 'BOT_UPDATE'
 export const PUBLISH = 'BOT_PUBLISH'
+export const PUBLISH_SUCCESS = 'BOT_PUBLISH_SUCCESS'
 export const UNPUBLISH = 'BOT_UNPUBLISH'
+export const UNPUBLISH_SUCCESS = 'BOT_UNPUBLISH_SUCCESS'
 export const DELETE = 'BOT_DELETE'
 
+export const botUpdate = (bot : T.Bot) : T.BotAction => ({
```
Comment (bcardiff) [inline app/javascript/actions/bot.js]: I find a bit confusing to have botUpdate and updateBot. But I can't come with a better alterantive 🤷‍♂️ .

maybe not export them and/or add a `_` prefix?

File: app/javascript/actions/channels.js [hunk 3]
```diff
@@ -6,28 +6,34 @@ export const RECEIVE = 'CHANNELS_RECEIVE'
...
+export const channelsReceiveError = () : T.ChannelsAction => ({
+  type: RECEIVE_ERROR
+})
+
 export const fetchChannels = (scope : {botId : number}) => (dispatch : T.Dispatch, getState : T.GetState) => {
   const state = getState()
 
   // Don't fetch Channels if they are already being fetched
-  if ((state.channels.fetching : boolean)) {
+  if (state.channels.fetching) {
```
Comment (bcardiff) [inline app/javascript/actions/channels.js]: 👎  ditto

File: app/services/gather_bot_stats.rb [hunk 4]
```diff
@@ -9,9 +9,9 @@ def self.run(bot, period = 'today')
 
     skills = bot.skills.index_by(&:id)
     {
-      active_users: summary['active_users'],
-      messages_sent: summary['messages_sent'],
-      messages_received: summary['messages_received'],
+      active_users: summary['active_users'] || 0,
```
Comment (bcardiff) [inline app/services/gather_bot_stats.rb]: NB: This is fixed in the engine already. I am not sure we need to be defensive here. up to you.

File: app/javascript/actions/bot.js (hunk 2 above)
Comment (ggiraldez) [reply app/javascript/actions/bot.js]: I renamed all direct action creators after the constants of their types. The other actions were there before. One of the reasons for explicitly implementing the action creators (as opposed to just `dispatch({type: ...})` is for testing, so not exporting them defeats the purpose (documentation being the other main one IMHO). I'll settle for the `_` prefix :)

File: app/javascript/actions/bots.js (hunk 1 above)
Comment (ggiraldez) [reply app/javascript/actions/bots.js]: Ugh... I should have asked why you were making the explicit cast.

File: app/services/gather_bot_stats.rb (hunk 4 above)
Comment (ggiraldez) [reply app/services/gather_bot_stats.rb]: It doesn't hurt IMO.

**Pull request merged**